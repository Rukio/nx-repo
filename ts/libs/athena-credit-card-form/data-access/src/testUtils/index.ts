export * from './setupTestStore';
export * from './helpers';
