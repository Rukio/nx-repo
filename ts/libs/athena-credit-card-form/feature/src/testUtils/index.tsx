import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { authSlice } from '@*company-data-covered*/auth0/data-access';
import {
  athenaCreditCardFormApiSlice,
  collectPaymentSlice,
  creditCardSlice,
  RootState,
} from '@*company-data-covered*/athena-credit-card-form/data-access';
import { setupListeners } from '@reduxjs/toolkit/query';
import { Provider } from 'react-redux';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'queries'>;
  userEventOptions?: Omit<UserEventSetupConfig, 'delay'>;
  withRouter?: boolean;
  routerProps?: MemoryRouterProps;
  preloadedState?: ConfigureStoreOptions<RootState>['preloadedState'];
};

const customRender = (
  ui: ReactElement,
  options: CustomRenderOptions = {
    withRouter: true,
  }
) => {
  const {
    withRouter,
    routerProps = {},
    userEventOptions,
    renderOptions,
    preloadedState,
  } = options;

  const store = configureStore({
    reducer: {
      [authSlice.name]: authSlice.reducer,
      [athenaCreditCardFormApiSlice.reducerPath]:
        athenaCreditCardFormApiSlice.reducer,
      [collectPaymentSlice.name]: collectPaymentSlice.reducer,
      [creditCardSlice.name]: creditCardSlice.reducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        immutableCheck: false,
        serializableCheck: false,
      }).concat(athenaCreditCardFormApiSlice.middleware),
    preloadedState,
  });

  setupListeners(store.dispatch);

  const component = withRouter ? (
    <MemoryRouter {...routerProps}>{ui}</MemoryRouter>
  ) : (
    ui
  );

  return {
    user: userEvent.setup({ ...userEventOptions, delay: null }),
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
