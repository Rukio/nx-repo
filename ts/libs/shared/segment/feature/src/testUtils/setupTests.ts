import '@*company-data-covered*/shared/testing/react/setup';
