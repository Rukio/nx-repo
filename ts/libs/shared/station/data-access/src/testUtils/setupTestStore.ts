import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { authSlice, AUTH_FEATURE_KEY } from '@*company-data-covered*/auth0/data-access';
import { stationApiSlice } from '../';

export const setupTestStore = () =>
  configureStore({
    reducer: combineReducers({
      [AUTH_FEATURE_KEY]: authSlice.reducer,
      [stationApiSlice.reducerPath]: stationApiSlice.reducer,
    }),
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(stationApiSlice.middleware),
  });
