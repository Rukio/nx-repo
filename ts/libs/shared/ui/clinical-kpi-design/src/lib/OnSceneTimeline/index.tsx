import { FC } from 'react';
import { Box, Grid, makeSxStyles } from '@*company-data-covered*/design-system';
import {
  calculateDisplayTimeRange,
  getActionColor,
  getTimeLabelLeftOffset,
} from './utilities';
import { Typography } from '@mui/material';

export enum Event {
  EnRoute = 'En Route',
  OnScene = 'On Scene',
  Break = 'Break',
  Idle = 'Idle',
}

export interface OnSceneTimeLineEvent {
  type: Event;
  startTime: string;
  endTime: string;
}

interface OnSceneTimeLineProps {
  events: OnSceneTimeLineEvent[];
}

export const TIME_LABEL_TEST_ID = 'time-label-test-id';
export const LEGEND_EVENT_TEST_ID = (event: Event) => `legend-${event}-test-id`;

const makeStyles = () =>
  makeSxStyles({
    root: { position: 'relative', height: '20px' },
    timeMark: {
      position: 'absolute',
      top: '-15px',
      width: '30px',
      textAlign: 'center',
      fontSize: '12px',
    },
    timeMarkContainer: {
      position: 'absolute',
      top: '0',
      width: '1px',
      height: '100%',
      color: 'text.secondary',
      backgroundColor: 'grey.200',
    },
    action: {
      position: 'absolute',
      top: '0',
      height: '100%',
    },
    legendContainer: {
      display: 'flex',
      gap: 2,
      position: 'absolute',
      top: '150%',
    },
    legendEvent: {
      display: 'flex',
      alignItems: 'center',
      gap: 0.5,
    },
    legendCircle: {
      borderRadius: '50%',
      width: 12,
      height: 12,
    },
  });

const OnSceneTimeline: FC<OnSceneTimeLineProps> = ({ events }) => {
  const styles = makeStyles();

  const firstEvent = events.at(0);
  const lastEvent = events.at(-1);

  if (!firstEvent || !lastEvent) {
    return null;
  }

  const { minTimeInMs, maxTimeInMs } = calculateDisplayTimeRange(
    firstEvent,
    lastEvent
  );

  const timeRangeInMs = Math.abs(maxTimeInMs - minTimeInMs);

  const renderTimeLabels = () => {
    const timeLabels: JSX.Element[] = [];
    const currentTime = new Date(minTimeInMs);

    while (currentTime <= new Date(maxTimeInMs)) {
      const hour = currentTime.getHours();

      const currentTimeIsMs = currentTime.getTime();

      const containerLeft =
        ((currentTimeIsMs - minTimeInMs) / timeRangeInMs) * 100;

      const isFirstTimeLabel = currentTimeIsMs === minTimeInMs;
      const isLastTimeLabel = currentTimeIsMs === maxTimeInMs;

      const timeLeft = getTimeLabelLeftOffset(
        isFirstTimeLabel,
        isLastTimeLabel
      );

      timeLabels.push(
        <Box
          key={hour}
          data-testid={TIME_LABEL_TEST_ID}
          sx={[
            styles.timeMarkContainer,
            {
              left: `${containerLeft}%`,
            },
          ]}
        >
          <Box
            sx={[
              styles.timeMark,
              {
                left: `${timeLeft}px`,
              },
            ]}
          >
            {hour}:00
          </Box>
        </Box>
      );

      currentTime.setTime(currentTime.getTime() + 60 * 60 * 1000); // Add one hour
    }

    return timeLabels;
  };

  return (
    <Box sx={styles.root}>
      {renderTimeLabels()}
      {events.map((event, index) => {
        const { startTime, endTime, type } = event;

        const durationInMs =
          new Date(endTime).getTime() - new Date(startTime).getTime();

        const startMs = Math.floor(new Date(startTime).getTime() - minTimeInMs);

        const width = (durationInMs / timeRangeInMs) * 100;
        const left = (startMs / timeRangeInMs) * 100;

        return (
          <Box
            key={index}
            sx={[
              styles.action,
              {
                left: `${left}%`,
                width: `${width}%`,
                backgroundColor: getActionColor(type),
              },
            ]}
          />
        );
      })}
      <Grid sx={styles.legendContainer}>
        {Object.values(Event).map((event) => (
          <Grid
            key={event}
            sx={styles.legendEvent}
            data-testid={LEGEND_EVENT_TEST_ID(event)}
          >
            <Box
              sx={[
                styles.legendCircle,
                { backgroundColor: getActionColor(event) },
              ]}
            />
            <Typography variant="caption" color="text.secondary">
              {event}
            </Typography>
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};

export default OnSceneTimeline;
