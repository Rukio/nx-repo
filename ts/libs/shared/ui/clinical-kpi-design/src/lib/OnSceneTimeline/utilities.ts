import { Event, OnSceneTimeLineEvent } from './index';
import { theme } from '@*company-data-covered*/design-system';

const COLORS_FOR_EVENTS = {
  'En Route': theme.palette.info.main,
  'On Scene': theme.palette.info.light,
  Break: theme.palette.grey.A700,
  Idle: theme.palette.error.light,
};

const ceilRoundToHour = (date: Date) => {
  const MillisecondsInHours = 60 * 60 * 1000;

  return new Date(
    Math.ceil(date.getTime() / MillisecondsInHours) * MillisecondsInHours
  );
};

const floorRoundToHour = (date: string): number =>
  new Date(date).setMinutes(0, 0, 0);

export const getActionColor = (type: Event) => COLORS_FOR_EVENTS[type];

export const calculateDisplayTimeRange = (
  firstAction: OnSceneTimeLineEvent,
  lastAction: OnSceneTimeLineEvent
) => {
  const minTimeInMs = floorRoundToHour(firstAction.startTime);
  const maxTimeInMs = ceilRoundToHour(new Date(lastAction.endTime)).getTime();

  return { minTimeInMs, maxTimeInMs };
};

export const getTimeLabelLeftOffset = (
  isFirstLabel: boolean,
  isLastLabel: boolean
): string => {
  if (isFirstLabel) {
    return '0';
  }

  if (isLastLabel) {
    return '-30';
  }

  return '-15';
};
