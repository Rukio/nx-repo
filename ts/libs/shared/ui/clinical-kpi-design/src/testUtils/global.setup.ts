module.exports = async () => {
  process.env.TZ = process.env.TZ || 'America/Denver';
};
