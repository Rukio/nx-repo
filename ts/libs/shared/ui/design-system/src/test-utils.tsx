import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
