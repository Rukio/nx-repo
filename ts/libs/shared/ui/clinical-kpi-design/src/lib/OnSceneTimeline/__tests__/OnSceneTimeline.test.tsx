import { render, screen } from '../../../testUtils/test-utils';
import { BasicOnSceneTimeLine } from '../__storybook__/OnSceneTimeline.stories';
import {
  calculateDisplayTimeRange,
  getTimeLabelLeftOffset,
} from '../utilities';
import {
  OnSceneTimeLineEvent,
  Event,
  TIME_LABEL_TEST_ID,
  LEGEND_EVENT_TEST_ID,
} from '../index';

const getMillisecondsFromDateString = (date: string): number =>
  new Date(date).getTime();

describe('Unit tests', () => {
  describe('getTimeLabelLeftOffset()', () => {
    it('should return 0 if isFirstTimeLabel is true', () => {
      const actualResult = getTimeLabelLeftOffset(true, false);
      expect(actualResult).toEqual('0');
    });

    it('should return -30 if isLastLabel is true', () => {
      const actualResult = getTimeLabelLeftOffset(false, true);
      expect(actualResult).toEqual('-30');
    });

    it('should return -15 if isFirstTimeLabel and isLastLabel are false', () => {
      const actualResult = getTimeLabelLeftOffset(false, false);
      expect(actualResult).toEqual('-15');
    });
  });

  describe('calculateDisplayTimeRange()', () => {
    it('should return correct values when first event start at X hours sharp', () => {
      const firstEvent: OnSceneTimeLineEvent = {
        type: Event.EnRoute,
        startTime: '2023-05-23T10:00:00',
        endTime: '2023-05-23T10:40:00',
      };
      const expectedMinTimeDate = '2023-05-23T10:00:00';

      const lastEvent: OnSceneTimeLineEvent = {
        type: Event.OnScene,
        startTime: '2023-05-23T10:40:00',
        endTime: '2023-05-23T12:20:00',
      };
      const expectedMaxTimeDate = '2023-05-23T13:00:00';

      const { minTimeInMs: actualMinTime, maxTimeInMs: actualMaxTime } =
        calculateDisplayTimeRange(firstEvent, lastEvent);

      expect(actualMinTime).toEqual(
        getMillisecondsFromDateString(expectedMinTimeDate)
      );
      expect(actualMaxTime).toEqual(
        getMillisecondsFromDateString(expectedMaxTimeDate)
      );
    });

    it('should return correct values when first event start at X hours and Y minutes', () => {
      const firstEvent: OnSceneTimeLineEvent = {
        type: Event.EnRoute,
        startTime: '2023-05-23T10:38:00',
        endTime: '2023-05-23T10:52:00',
      };
      const expectedMinTimeDate = '2023-05-23T10:00:00';

      const lastEvent: OnSceneTimeLineEvent = {
        type: Event.OnScene,
        startTime: '2023-05-23T10:52:00',
        endTime: '2023-05-23T12:20:00',
      };
      const expectedMaxTimeDate = '2023-05-23T13:00:00';

      const { minTimeInMs: actualMinTime, maxTimeInMs: actualMaxTime } =
        calculateDisplayTimeRange(firstEvent, lastEvent);

      expect(actualMinTime).toEqual(
        getMillisecondsFromDateString(expectedMinTimeDate)
      );
      expect(actualMaxTime).toEqual(
        getMillisecondsFromDateString(expectedMaxTimeDate)
      );
    });
  });
});

describe('Integration tests', () => {
  test('snapshot tests', () => {
    const { asFragment } = render(<BasicOnSceneTimeLine />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render correct amount of time labels', () => {
    render(<BasicOnSceneTimeLine />);

    const timeLabels = screen.getAllByTestId(TIME_LABEL_TEST_ID);

    expect(timeLabels.length).toEqual(10);

    timeLabels.forEach((timeLabel) => expect(timeLabel).toBeInTheDocument());
  });

  test('should render correct amount of legend circles', () => {
    render(<BasicOnSceneTimeLine />);

    Object.values(Event).forEach((legendEvent) => {
      const legendCircle = screen.getByTestId(
        LEGEND_EVENT_TEST_ID(legendEvent)
      );

      expect(legendCircle).toBeInTheDocument();
    });
  });
});
