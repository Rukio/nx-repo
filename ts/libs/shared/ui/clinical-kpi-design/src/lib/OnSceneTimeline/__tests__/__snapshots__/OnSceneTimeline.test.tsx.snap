// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests snapshot tests 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  height: 20px;
}

.emotion-1 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 0%;
}

.emotion-2 {
  position: absolute;
  top: -15px;
  width: 30px;
  text-align: center;
  font-size: 12px;
  left: 0px;
}

.emotion-3 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 11.11111111111111%;
}

.emotion-4 {
  position: absolute;
  top: -15px;
  width: 30px;
  text-align: center;
  font-size: 12px;
  left: -15px;
}

.emotion-5 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 22.22222222222222%;
}

.emotion-7 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 33.33333333333333%;
}

.emotion-9 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 44.44444444444444%;
}

.emotion-11 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 55.55555555555556%;
}

.emotion-13 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 66.66666666666666%;
}

.emotion-15 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 77.77777777777779%;
}

.emotion-17 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 88.88888888888889%;
}

.emotion-19 {
  position: absolute;
  top: 0;
  width: 1px;
  height: 100%;
  color: rgba(0, 0, 0, 0.6);
  background-color: #eeeeee;
  left: 100%;
}

.emotion-20 {
  position: absolute;
  top: -15px;
  width: 30px;
  text-align: center;
  font-size: 12px;
  left: -30px;
}

.emotion-21 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 0%;
  width: 7.4074074074074066%;
  background-color: #007C79;
}

.emotion-22 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 7.4074074074074066%;
  width: 18.51851851851852%;
  background-color: #00CCC7;
}

.emotion-23 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 25.925925925925924%;
  width: 3.7037037037037033%;
  background-color: #FF9E84;
}

.emotion-24 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 29.629629629629626%;
  width: 3.7037037037037033%;
  background-color: #007C79;
}

.emotion-25 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 33.33333333333333%;
  width: 0.9259259259259258%;
  background-color: #FF9E84;
}

.emotion-26 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 34.25925925925926%;
  width: 10.185185185185185%;
  background-color: #00CCC7;
}

.emotion-27 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 44.44444444444444%;
  width: 5.555555555555555%;
  background-color: #00CCC7;
}

.emotion-28 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 50%;
  width: 5.555555555555555%;
  background-color: #3A5C74;
}

.emotion-29 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 55.55555555555556%;
  width: 1.8518518518518516%;
  background-color: #FF9E84;
}

.emotion-30 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 57.407407407407405%;
  width: 5.555555555555555%;
  background-color: #007C79;
}

.emotion-31 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 62.96296296296296%;
  width: 3.7037037037037033%;
  background-color: #00CCC7;
}

.emotion-32 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 66.66666666666666%;
  width: 2.7777777777777777%;
  background-color: #FF9E84;
}

.emotion-33 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 69.44444444444444%;
  width: 8.333333333333332%;
  background-color: #007C79;
}

.emotion-34 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 77.77777777777779%;
  width: 19.444444444444446%;
  background-color: #00CCC7;
}

.emotion-35 {
  position: absolute;
  top: 0;
  height: 100%;
  left: 97.22222222222221%;
  width: 2.7777777777777777%;
  background-color: #FF9E84;
}

.emotion-36 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
  position: absolute;
  top: 150%;
}

.emotion-37 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
}

.emotion-38 {
  border-radius: 50%;
  width: 12px;
  height: 12px;
  background-color: #007C79;
}

.emotion-39 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  letter-spacing: 0.03333em;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-41 {
  border-radius: 50%;
  width: 12px;
  height: 12px;
  background-color: #00CCC7;
}

.emotion-44 {
  border-radius: 50%;
  width: 12px;
  height: 12px;
  background-color: #3A5C74;
}

.emotion-47 {
  border-radius: 50%;
  width: 12px;
  height: 12px;
  background-color: #FF9E84;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        10:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-3"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        11:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-5"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        12:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-7"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        13:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-9"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        14:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-11"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        15:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-13"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        16:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-15"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        17:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-17"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-4"
      >
        18:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-19"
      data-testid="time-label-test-id"
    >
      <div
        class="MuiBox-root emotion-20"
      >
        19:00
      </div>
    </div>
    <div
      class="MuiBox-root emotion-21"
    />
    <div
      class="MuiBox-root emotion-22"
    />
    <div
      class="MuiBox-root emotion-23"
    />
    <div
      class="MuiBox-root emotion-24"
    />
    <div
      class="MuiBox-root emotion-25"
    />
    <div
      class="MuiBox-root emotion-26"
    />
    <div
      class="MuiBox-root emotion-27"
    />
    <div
      class="MuiBox-root emotion-28"
    />
    <div
      class="MuiBox-root emotion-29"
    />
    <div
      class="MuiBox-root emotion-30"
    />
    <div
      class="MuiBox-root emotion-31"
    />
    <div
      class="MuiBox-root emotion-32"
    />
    <div
      class="MuiBox-root emotion-33"
    />
    <div
      class="MuiBox-root emotion-34"
    />
    <div
      class="MuiBox-root emotion-35"
    />
    <div
      class="MuiGrid-root emotion-36"
    >
      <div
        class="MuiGrid-root emotion-37"
        data-testid="legend-En Route-test-id"
      >
        <div
          class="MuiBox-root emotion-38"
        />
        <span
          class="MuiTypography-root MuiTypography-caption emotion-39"
        >
          En Route
        </span>
      </div>
      <div
        class="MuiGrid-root emotion-37"
        data-testid="legend-On Scene-test-id"
      >
        <div
          class="MuiBox-root emotion-41"
        />
        <span
          class="MuiTypography-root MuiTypography-caption emotion-39"
        >
          On Scene
        </span>
      </div>
      <div
        class="MuiGrid-root emotion-37"
        data-testid="legend-Break-test-id"
      >
        <div
          class="MuiBox-root emotion-44"
        />
        <span
          class="MuiTypography-root MuiTypography-caption emotion-39"
        >
          Break
        </span>
      </div>
      <div
        class="MuiGrid-root emotion-37"
        data-testid="legend-Idle-test-id"
      >
        <div
          class="MuiBox-root emotion-47"
        />
        <span
          class="MuiTypography-root MuiTypography-caption emotion-39"
        >
          Idle
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
